File: packages/ipc/src/Device.ts
import type { DeviceInfo, Device as IDevice } from "@cortex/platform"
import { invoke } from "@tauri-apps/api/core"

export class Device implements IDevice {
	async getDeviceId(): Promise<string> {
		return await invoke<string>("get_device_id")
	}

	async getDeviceInfo(): Promise<DeviceInfo> {
		return await invoke<DeviceInfo>("get_device_info")
	}
}

File: apps/desktop/src-tauri/src/commands/device.rs
use crate::device::{self, DeviceInfo};

#[tauri::command]
pub fn get_device_id() -> Result<String, String> {
    device::get_device_id()
}

#[tauri::command]
pub fn get_device_info() -> Result<DeviceInfo, String> {
    device::get_device_info()
}
